idf_build_apps/manifest/manifest.py

```diff
@@ -35,14 +35,20 @@ class IfClause:
     def __init__(self, stmt: str, temporary: bool = False, reason: t.Optional[str] = None) -> None:
         try:
             self.stmt: BoolStmt = BOOL_EXPR.parseString(stmt)[0]
-        except ParseException:
-            raise InvalidIfClause(f'Invalid if statement: {stmt}')
+        except (ParseException, InvalidIfClause) as ex:
+            raise InvalidIfClause(f'Invalid if clause: {stmt}. {ex}')
 
         self.temporary = temporary
         self.reason = reason
 
         if self.temporary is True and not self.reason:
-            raise InvalidIfClause('"reason" must be set when "temporary: true"')
+            raise InvalidIfClause(
+                f'Invalid if clause "{stmt}". '
+                f'"reason" must be set when "temporary: true".'
+                f'For example:'
+                f'   temporary: true'
+                f'   reason: lack of ci runners'
+            )
```

Comment:
the output is not nice. there's no newline added between each lines. so the output would be 

```
Invalid manifest file .build-test-rules.yml: Invalid if clause ""esp32" == "1" and FOO == 1". "reason" must be set when "temporary: true".For example:   temporary: true   reason: lack of ci runners
```

Reply:
```suggestion
            raise InvalidIfClause(
                f'Invalid if clause "{stmt}". '
                f'"reason" must be set when "temporary: true". '
                f'For example:\n'
                f'  - if: {stmt}\n'
                f'    temporary: true\n'
                f'    reason: lack of ci runners\n'
            )
```